src/extension.ts

```diff
@@ -139,6 +143,19 @@ function handleVisibleEditorChanges(event) {
 	fileToNode.delete(k);
 }
 
+function startFileSystemWatcher(): void {
+	fileSystemWatcher = vscode.workspace.createFileSystemWatcher('**/*.yaml');
+	fileSystemWatcher.onDidCreate( () => {
+		dataVirtProvider.refresh();
+	});
+	fileSystemWatcher.onDidDelete( () => {
+		dataVirtProvider.refresh();
+	});
+	fileSystemWatcher.onDidChange( () => {
+		dataVirtProvider.refresh();
+	});
```

Comment:
I fear that a refresh for all files on the whole tree for all changes will cause troubles with big workspaces.

I think we should try with a relatively medium workspace at least (50 yaml files?) and see the impact. (or maybe you already tried?)
If nto noticeable, we migth go like that and create an enhancement request for later on real big workspaces. if it is causing delay, I think we should improve it within this PR

Reply:
@rareddy What do you think? How many yaml files will the user have in the workspace?

Reply:
I will create a test environment with at least 50 files. That said after discussion with Ramesh it turned out that the yaml files can be really huge. So we need to refactor the treemodel and logic anyways to ensure we are performing good enough. Raised https://issues.redhat.com/browse/FUSETOOLS2-344 for this purpose.

Reply:
Tested with 50 yaml files and there was no difference in speed recognizable. That might become an issue later but for now I don't see this as a blocker.